### PR TITLE
Added default sorting milestones by name

### PR DESCRIPTION
**Pull request by BElluu**
#26996 
Added default sorting for milestones by name.
Additional, name for sorting closestduedate and furthestduedate was broken, so I fixed it.

### CODE REVIEW

File: models/issues/milestone_list.go [hunk 1]
```diff
@@ -78,8 +78,10 @@ func GetMilestones(opts GetMilestonesOption) (MilestoneList, int64, error) {
 		sess.Desc("num_issues")
 	case "id":
 		sess.Asc("id")
+	case "name":
+		sess.Desc("name")
 	default:
-		sess.Asc("deadline_unix").Asc("id")
+		sess.Asc("name").Asc("id")
```
Comment (delvh) [inline models/issues/milestone_list.go]: I am against that change.
That is not how it should behave.
What I can agree on, however, is
```suggestion
		sess.Asc("deadline_unix").Asc("name")
```

File: models/issues/milestone_list.go [hunk 2]
```diff
@@ -124,8 +126,10 @@ func SearchMilestones(repoCond builder.Cond, page int, isClosed bool, sortType,
 		sess.Asc("num_issues")
 	case "mostissues":
 		sess.Desc("num_issues")
+	case "name":
+		sess.Asc("name")
 	default:
-		sess.Asc("deadline_unix")
+		sess.Asc("name")
```
Comment (delvh) [inline models/issues/milestone_list.go]:
```suggestion
		sess.Asc("deadline_unix").Asc("name")
```

File: templates/repo/issue/milestones.tmpl [hunk 3]
```diff
@@ -40,7 +40,8 @@
 					{{svg "octicon-triangle-down" 14 "dropdown icon"}}
 				</span>
 				<div class="menu">
-					<a class="{{if or (eq .SortType "closestduedate") (not .SortType)}}active {{end}}item" href="{{$.Link}}?sort=closestduedate&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.earliest_due_data"}}</a>
+					<a class="{{if or (eq .SortType "name") (not .SortType)}}active {{end}}item" href="{{$.Link}}?sort=name={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.name"}}</a>
+					<a class="{{if eq .SortType "closestduedate"}}active {{end}}item" href="{{$.Link}}?sort=closestduedate&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.earliest_due_data"}}</a>
```
Comment (delvh) [inline templates/repo/issue/milestones.tmpl]: I still disagree with changing the default sort.
```suggestion
					<a class="{{if eq .SortType "name") (not .SortType)}}active {{end}}item" href="{{$.Link}}?sort=name={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.name"}}</a>
					<a class="{{if or (eq .SortType "closestduedate") (not .SortType)}}active {{end}}item" href="{{$.Link}}?sort=closestduedate&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.earliest_due_data"}}</a>
```

File: templates/user/dashboard/milestones.tmpl [hunk 4]
```diff
@@ -62,8 +62,9 @@
 							{{svg "octicon-triangle-down" 14 "dropdown icon"}}
 						</span>
 						<div class="menu">
-							<a class="{{if or (eq .SortType "closestduedate") (not .SortType)}}active {{end}}item" href="{{$.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort=closestduedate&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.closest_due_date"}}</a>
-							<a class="{{if eq .SortType "furthestduedate"}}active {{end}}item" href="{{$.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort=furthestduedate&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.furthest_due_date"}}</a>
+							<a class="{{if or (eq .SortType "name") (not .SortType)}}active {{end}}item" href="{{$.Link}}?sort=name={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.name"}}</a>
+							<a class="{{if eq .SortType "closestduedate"}}active {{end}}item" href="{{$.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort=closestduedate&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.earliest_due_data"}}</a>
```
Comment (delvh) [inline templates/user/dashboard/milestones.tmpl]:
```suggestion
							<a class="{{if eq .SortType "name"}}active {{end}}item" href="{{$.Link}}?sort=name={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.name"}}</a>
							<a class="{{if or (eq .SortType "closestduedate") (not .SortType)}}active {{end}}item" href="{{$.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort=closestduedate&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.earliest_due_data"}}</a>
```

**Review by delvh — CHANGES_REQUESTED** [review]

File: models/issues/milestone_list.go (hunk 1 above)
Comment (techknowlogick) [reply models/issues/milestone_list.go]: I am unsure about sorting by name as you run into `v1 ... v19 v2 v20` trouble as it's not aware of version numbers.

**Comment by BElluu** [comment]
Hello :)
I reverted default sort, but still able to choose sorting by name :)
Thank you for your time!

**Review by delvh — APPROVED** [review]

File: templates/repo/issue/milestones.tmpl [hunk 5]
```diff
@@ -46,6 +46,7 @@
 					<a class="{{if eq .SortType "mostcomplete"}}active {{end}}item" href="{{$.Link}}?sort=mostcomplete&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.most_complete"}}</a>
 					<a class="{{if eq .SortType "mostissues"}}active {{end}}item" href="{{$.Link}}?sort=mostissues&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.most_issues"}}</a>
 					<a class="{{if eq .SortType "leastissues"}}active {{end}}item" href="{{$.Link}}?sort=leastissues&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.least_issues"}}</a>
+					<a class="{{if eq .SortType "name"}}active {{end}}item" href="{{$.Link}}?sort=name={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.name"}}</a>
```
Comment (lunny) [inline templates/repo/issue/milestones.tmpl]:
```suggestion
					<a class="{{if eq .SortType "name"}}active {{end}}item" href="{{$.Link}}?sort=namez&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.name"}}</a>
```

File: templates/user/dashboard/milestones.tmpl [hunk 6]
```diff
@@ -62,12 +62,13 @@
 							{{svg "octicon-triangle-down" 14 "dropdown icon"}}
 						</span>
 						<div class="menu">
-							<a class="{{if or (eq .SortType "closestduedate") (not .SortType)}}active {{end}}item" href="{{$.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort=closestduedate&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.closest_due_date"}}</a>
-							<a class="{{if eq .SortType "furthestduedate"}}active {{end}}item" href="{{$.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort=furthestduedate&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.furthest_due_date"}}</a>
+							<a class="{{if or (eq .SortType "closestduedate") (not .SortType)}}active {{end}}item" href="{{$.Link}}?repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]&sort=closestduedate&state={{$.State}}&q={{$.Keyword}}">{{.locale.Tr "repo.milestones.filter_sort.earliest_due_data"}}</a>
```
Comment (lunny) [inline templates/user/dashboard/milestones.tmpl]: Unrelated changes?

Comment (delvh) [reply templates/user/dashboard/milestones.tmpl]: Yes, but if I see it correctly, the translation is currently either not present or the wrong one

File: templates/repo/issue/milestones.tmpl (hunk 5 above)
Comment (lunny) [reply templates/repo/issue/milestones.tmpl]: Sorry, I made a typo.

**Comment by 6543** [comment]
Its fine but the conflicts need to be solved

**Comment by BElluu** [comment]
Ouh PR last year :D I will fix this conflicts tomorrow :)

**Comment by 6543** [comment]
sorry somethimes it gets lost till one picks it up again :sweat:

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
Just address tje lint issue and its good to go

**Comment by BElluu** [comment]
I hope will be ok now, because I'm away from my PC with IDE, so I used edit file in github :)
